### PR TITLE
Use context instead of with_context

### PR DESCRIPTION
**Pull request by mo8it**
Replace `.with_context(|| "…")` with `.context("…")` to avoid calling a trivial closure. It is also shorter :)

On the other hand, use `.with_context(|| format!(…))` instead of `.context(format!(…))` to avoid unneeded string allocation.

### CODE REVIEW

File: crates/rustfix/tests/parse_and_replace.rs [hunk 1]
```diff
@@ -164,8 +164,8 @@ fn test_rustfix_with_file<P: AsRef<Path>>(file: P, mode: &str) -> Result<(), Err
 
     debug!("next up: {:?}", file);
     let code = fs::read_to_string(file)?;
-    let errors =
-        compile_and_get_json_errors(file).context(format!("could compile {}", file.display()))?;
+    let errors = compile_and_get_json_errors(file)
+        .with_context(|| format!("could not compile {}", file.display()))?;
```
Comment (mo8it) [inline crates/rustfix/tests/parse_and_replace.rs:168]: This is the only non-trivial change. This should be "could not", right?

**Review by weihanglo — COMMENTED** [review]
No objection to these changes. Thank you!

File: src/cargo/core/registry.rs [hunk 2]
```diff
@@ -444,8 +444,8 @@ impl<'gctx> PackageRegistry<'gctx> {
                         patches must point to different sources",
                         dep.package_name(),
                         url
-                    ))
-                    .context(format!("failed to resolve patches for `{}`", url));
+                    )
+                    .context(format!("failed to resolve patches for `{}`", url)));
```
[the remark this answers is not in the document]
Comment (mo8it) [reply src/cargo/core/registry.rs:448]: No, we use [`anyhow::Error::context`](https://docs.rs/anyhow/latest/anyhow/struct.Error.html#method.context) here on `anyhow::Error`, not [`anyhow::Context::context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.context) on a `Result`. This is error is built and returned, not conditionally. In fact, `with_context` doesn't exist for `anyhow::Error`.

[the remark this answers is not in the document]
Comment (weihanglo) [reply src/cargo/core/registry.rs:448]: I see. Misunderstood. Thanks for the clarification!

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
@bors r+

Thanks!

**Pull request merged**